packages/react/dropdown-menu/src/DropdownMenu.tsx

```diff
@@ -146,37 +182,49 @@ const DropdownMenuContent = React.forwardRef((props, forwardedRef) => {
   } = props;
   const context = useDropdownMenuContext(CONTENT_NAME);
   return (
-    <SubmenuContext.Provider value={true}>
-      <MenuPrimitive.Content
-        id={context.contentId}
-        {...contentProps}
-        ref={forwardedRef}
-        disableOutsidePointerEvents={disableOutsidePointerEvents}
-        disableOutsideScroll={disableOutsideScroll}
-        portalled={portalled}
-        style={{
-          ...props.style,
-          // re-namespace exposed content custom property
-          ['--radix-dropdown-menu-content-transform-origin' as any]: 'var(--radix-popper-transform-origin)',
-        }}
-        trapFocus
-        onCloseAutoFocus={composeEventHandlers(onCloseAutoFocus, (event) => {
-          event.preventDefault();
-          context.triggerRef.current?.focus();
-        })}
-        onPointerDownOutside={composeEventHandlers(
-          props.onPointerDownOutside,
-          (event) => {
-            const target = event.target as HTMLElement;
-            const targetIsTrigger = context.triggerRef.current?.contains(target);
-            // prevent dismissing when clicking the trigger
-            // as it's already setup to close, otherwise it would close and immediately open.
-            if (targetIsTrigger) event.preventDefault();
-          },
-          { checkForDefaultPrevented: false }
-        )}
-      />
-    </SubmenuContext.Provider>
+    <ContentContext.Provider value={true}>
+      {context.isSubmenu ? (
+        <MenuPrimitive.Content
+          {...contentProps}
+          ref={forwardedRef}
+          disableOutsidePointerEvents={disableOutsidePointerEvents}
+          disableOutsideScroll={disableOutsideScroll}
+          portalled={portalled}
+          onCloseAutoFocus={onCloseAutoFocus}
+        />
+      ) : (
+        <MenuPrimitive.Content
+          id={context.contentId}
+          aria-labelledby={context.triggerId}
+          {...contentProps}
+          ref={forwardedRef}
+          disableOutsidePointerEvents={disableOutsidePointerEvents}
+          disableOutsideScroll={disableOutsideScroll}
+          portalled={portalled}
+          style={{
+            ...props.style,
+            // re-namespace exposed content custom property
+            ['--radix-dropdown-menu-content-transform-origin' as any]: 'var(--radix-popper-transform-origin)',
```

Comment:
Are we sure we don't also need this for animation on submenu content?

Reply:
Yeah we do, I think I'll this bit with the `commonProps` approach like I suggested above (https://github.com/radix-ui/primitives/pull/692#discussion_r648042127), it'll be clearer what the differences are and should be.

Reply:
Ah, I originally removed this because the `side` is always hardcoded for submenus but didn't think about collision detection! Will move this onto both.